dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/DynamicSpinLoopDetection.java

```diff
@@ -73,7 +73,8 @@ private List<LoopData> computeLoopData(Function func, LoopAnalysis loopAnalysis)
     }
 
     private void collectSideEffects(LoopData loop) {
-        if (loop.getStart().getSuccessor() instanceof SpinStart) {
+        // Loop shape is: loopStart -> Calling intrinsic __VERIFIER_spin_start -> #__VERIFIER_spin_start
+        if (loop.getStart().getSuccessors().get(2) instanceof SpinStart) {
```

Comment:
Just use `getSuccessor().getSuccessor()`. `getSuccessors` constructs the list of all successor events which feels like a waste if you only want to access the second one.